### ISSUE TITLE
Feature Request: Images over REST as URL

### ISSUE THREAD

**Issue by MatthewBooth** [pimcore/pimcore#1268]
# Feature Request 
## Please make sure all the following boxes are checked before submitting your feature request - thank you!
- [ ] There is no existing issue regarding the same topic

## Feature description 

Ability (option) to return image assets over the REST service as a URL, rather than base64 encoded image.

The reason being that some larger images become unworkable if using the REST API. This is mainly on image-heavy websites.

**Comment by weisswurstkanone** [comment]
Isn't that already possible by adding the "light" Parameter?
see https://www.pimcore.org/docs/latest/Web_Services/index.html#page_Get_Asset_By_ID

Get Asset By ID
Method: GET
URL: http://YOUR-DOMAIN/webservice/rest/asset/id/1281?apikey=[API-KEY]
**Optional Parameter: light if set to true, the response will not contain the data (base64) of the asset.**

**Comment by MatthewBooth** [comment]
Oh, that'll teach em to FULLY read the docs.

However, it now returns this:

```json
{
  "success": true,
  "data": {
    "id": 2,
    "parentId": 1,
    "type": "image",
    "filename": "logo-social-36b145d019f3628f5207919ea17dd71f.png",
    "path": "/",
    "mimetype": "image/png",
    "creationDate": 1486661437,
    "modificationDate": 1486661437,
    "userOwner": "2",
    "userModification": "2",
    "properties": null,
    "customSettings": {
      "imageWidth": 400,
      "imageHeight": 400,
      "imageDimensionsCalculated": true
    },
    "metadata": [],
    "checksum": {
      "algo": "sha1",
      "value": "535e90b338dcf9252bfb0a4ce2d1530ca1a1fd92"
    }
  }
}```

No URL to be found. Wouldn't be an issue as much but I'm using Amazon S3 as detailed [here](https://www.pimcore.org/docs/latest/Installation_and_Upgrade/System_Setup_and_Hosting/Amazon_AWS_Setup/Amazon_AWS_S3_Setup.html).

**Comment by weisswurstkanone** [comment]
Download URLs are typically assembled as follows:
                        $uri = $protocol . self::$host . "/website/var/assets" . $path . $filename;
We didn't see a need to add this URL to the response although we could probably simply add it.

**Comment by MatthewBooth** [comment]
Yeah, probably wouldn't be an issue at all if I wasn't using that Amazon S3 guide to have assets etc on S3 rather than the server.

Perhaps using Flysystem would help abstract this a little such that you would only need the filename (key) to get the URL of the image anyway? (that'd be something I'd happily add myself rather than needing you guys to implement a third party library in the whole project, of course.)

Going to end up in a few API calls per page to get images, however. The way I'm seeing it now, a page with 4 images will require 5 calls to get the document and images/assets. Am I getting that right or is there something else I'm missing?

**Comment by salexfo** [comment]
+1 to this feature request

**Comment by mathieu2011** [comment]
+1 for this feature request

We are using PimCore as headless CMS with a similar setup as the OP.
Would be nice to have the uri coming from the REST API instead of having to build it myself everywhere I want to access it.
